File: docs/code_samples/indexing/c#/automatic_grouping.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;
using System.IO;
using System.Net;

var sessionId = "66878f5e-c609-4796-9fc2-ecc6ae377cac";
var baseUrl = "http://dev-api.v2.areal.ai/api/v2";

// Assume client is already authenticated
var handler = new HttpClientHandler
{
    UseCookies = true,
    CookieContainer = new CookieContainer()
};
var client = new HttpClient(handler);

var downloadUrl = $"{baseUrl}/sessions/{sessionId}/download/?group_by=index_id";
var response = await client.PostAsync(downloadUrl, null);
response.EnsureSuccessStatusCode();

var content = await response.Content.ReadAsByteArrayAsync();
await File.WriteAllBytesAsync("documents.zip", content);

Console.WriteLine("✅ Zip file downloaded and saved as documents.zip");

File: docs/code_samples/indexing/python/automatic_grouping.py
import requests  # noqa

SESSION_ID = '66878f5e-c609-4796-9fc2-ecc6ae377cac'

response = client.post(
    f'{BASE_URL}/sessions/{SESSION_ID}/download/', params={'group_by': 'index_id'}
)
with open('documents.zip', 'wb') as f:
    f.write(response.content)

print('✅ Zip file downloaded and saved as documents.zip')
